BetterMD/elements/document.py

```
@@ -77,7 +134,45 @@ def get_elements_by_class_name(self, class_name: 'str'):
+        elif key.startswith("."):
+            return self.get_elements_by_class_name(key[1:])
+        else:
+            return self.get_elements_by_tag_name(key)
+
+    def get_by_text(self, text:'str'):
+        return self.text.get(text, [])
+
+    def get_by_attr(self, attr:'str', value:'str'):
+        return self.attrs.get(attr, {}).get(value, [])
+
+    def advanced_find(self, tag:'str', attrs:'dict[t.Literal["text"] | str, str | bool | int | float | tuple[str, str | bool | int | float] | list[str | bool | int | float | tuple[str, str | bool | int | float]]]' = {}):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)